### PR TITLE
Support null address in RegisterSignupRequest

### PR DESCRIPTION
**Pull request by racevedoo**
## Proposed changes

In our onboarding api, address is optional. In this lib, it was required.

This makes address nullable and adds proper testing

## Checklist
- [x] Style check and tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: src/main/java/com/incognia/onboarding/RegisterSignupRequest.java [hunk 1]
```diff
@@ -5,12 +5,13 @@
 import java.util.List;
 import lombok.Builder;
 import lombok.Value;
+import org.jetbrains.annotations.Nullable;
```
Comment (soareswallace) [inline src/main/java/com/incognia/onboarding/RegisterSignupRequest.java:8]: duvida: O import do nullable é esse mesmo?

File: src/test/java/com/incognia/api/IncogniaAPITest.java [hunk 2]
```diff
@@ -127,6 +127,62 @@ void testRegisterSignup_whenDataIsValid() {
     assertThat(signupAssessment.getReasons()).containsExactly(expectedReason);
   }
 
+  @Test
+  @DisplayName("should return the expected signup response")
```
Comment (soareswallace) [inline src/test/java/com/incognia/api/IncogniaAPITest.java:131]: nit: talvez ter um `displayName` nome que diferencie do teste da linha 68?

**Review by bcs5 — APPROVED** [review]

**Review by soareswallace — APPROVED** [review]

File: src/test/java/com/incognia/api/IncogniaAPITest.java (hunk 2 above)
Comment (racevedoo) [reply src/test/java/com/incognia/api/IncogniaAPITest.java:131]: sure, just forgot to change the name. fixing right now

File: src/main/java/com/incognia/onboarding/RegisterSignupRequest.java (hunk 1 above)
Comment (racevedoo) [reply src/main/java/com/incognia/onboarding/RegisterSignupRequest.java:8]: In this case it is, we do'nt use jakarta/javax here

**Review by soareswallace — APPROVED** [review]

File: src/test/java/com/incognia/api/IncogniaAPITest.java (hunk 2 above)
Comment (racevedoo) [reply src/test/java/com/incognia/api/IncogniaAPITest.java]: done

**Pull request merged**